internal/provider/vaultradar/resource_radar_source_github_enterprise.go

```diff
@@ -0,0 +1,235 @@
+
+	radar_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-radar/preview/2023-05-01/client/data_source_registration_service"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/provider/modifiers"
+)
+
+var (
+	_ resource.Resource              = &sourceGitHubEnterpriseResource{}
+	_ resource.ResourceWithConfigure = &sourceGitHubEnterpriseResource{}
+)
+
+func NewSourceGitHubEnterpriseResource() resource.Resource {
+	return &sourceGitHubEnterpriseResource{}
+}
+
+type sourceGitHubEnterpriseResource struct {
+	client *clients.Client
+}
+
+func (r *sourceGitHubEnterpriseResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vault_radar_source_github_enterprise"
+}
+
+func (r *sourceGitHubEnterpriseResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This Vault Radar resource manages HCP Cloud scans of GitHub Enterprise Server sources.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The ID of this resource.",
+			},
+			"domain_name": schema.StringAttribute{
+				Description: "Fully qualified domain name of the server. (Example: myserver.acme.com)",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^\S+$`),
+						"must not contain white space",
+					),
+				},
+			},
+			"github_organization": schema.StringAttribute{
+				Description: `GitHub organization Vault Radar will monitor. Example: "octocat" for the org https://yourcodeserver.com/octocat`,
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^[a-zA-Z0-9-_.]+$`),
+						"must contain only letters, numbers, hyphens, underscores, or periods",
+					),
+				},
+			},
+			"token": schema.StringAttribute{
+				Description: "GitHub personal access token.",
+				Required:    true,
+				Sensitive:   true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
```

Comment:
We don't need to destroy and recreate the datasource in Vault Radar to update the token right, is `RequiresReplace` the right modifier to use?

Reply:
I don't have the update implemented yet.
I will need to expose a public api for that.
I will add that in a future PR.

Reply:
Sounds good, I didn't realize that update wasn't implemented when adding the comment